kafka-rest/src/test/java/io/confluent/kafkarest/unit/PartitionsResourceBinaryProduceTest.java

```diff
@@ -255,4 +266,25 @@ public void testProduceInvalidRequest() {
       }
     }
   }
+
+  @Test
+  public void testProduceToPartitionAuthorizationErrors() {
+    for (TestUtils.RequestMediaType mediatype : TestUtils.V1_ACCEPT_MEDIATYPES) {
+      for (String requestMediatype : TestUtils.V1_REQUEST_ENTITY_TYPES_BINARY) {
+        Response
+            rawResponse =
+            produceToPartition(topicName, 0, mediatype.header, requestMediatype,
+                EmbeddedFormat.BINARY,
+                produceRecordsOnlyValues, produceKafkaAuthorizationExceptionResults);
+        assertOKResponse(rawResponse, mediatype.expected);
```

Comment:
Shouldn't this result in a 401?

Reply:
Updated the code to set Response status code  for  authz failures